src/uct/ib/rdmacm/rdmacm_cm.h

```diff
@@ -50,6 +56,36 @@ typedef struct uct_rdmacm_cm_config {
     double          timeout;
 } uct_rdmacm_cm_config_t;
 
+struct uct_rdmacm_cm_reserved_qpn_blk {
+    ucs_bitmap_t(256)       bmp;
+    uint32_t                first_qpn;
+    unsigned int            next_avail_qpn_offset;
+    ucs_list_link_t         entry;
+    struct mlx5dv_devx_obj  *obj;
+};
+
+struct uct_rdmacm_cm_reserved_qpn_context {
+     /* Avoid competed condition on the qpn resource for multi-threads */ 
+     pthread_mutex_t mutex;
```

Comment:
why not spinlock? it's more light weight

Reply:
@yosefe Changed.